lighthouse-cli/performance-experiment/server.js

```diff
@@ -84,26 +82,98 @@ function requestHandler(request, response) {
 }
 
 function reportRequestHandler(request, response) {
-  const html = reportGenerator.generateHTML(lhResults, 'cli');
-  response.writeHead(200, {'Content-Type': 'text/html'});
-  response.end(html);
+  try {
+    const id = request.parsedUrl.query.id || 0;
+    const results = database.getResults(id);
+
+    results.relatedReports = database.timeStamps.map((generatedTime, index) => {
+      return {reportUrl: `/?id=${index}`, url: database.url, generatedTime};
+    });
+
+    const reportGenerator = new ReportGenerator();
+    const html = reportGenerator.generateHTML(results, 'cli');
+
+    response.writeHead(200, {'Content-Type': 'text/html'});
+    response.end(html);
+  } catch (err) {
+    response.writeHead(404);
+    response.end('404: Resource Not Found');
+  }
 }
 
 function rerunRequestHandler(request, response) {
-  let message = '';
-  request.on('data', data => message += data);
-
-  request.on('end', () => {
-    const additionalFlags = JSON.parse(message);
-    const flags = Object.assign({}, lhParams.flags, additionalFlags);
-
-    lighthouse(lhParams.url, flags, lhParams.config).then(results => {
-      results.artifacts = undefined;
-      lhResults = results;
-      response.writeHead(200);
-      response.end();
+  try {
+    const flags = database.getFlags(request.parsedUrl.query.id || 0).flags;
+    let message = '';
+    request.on('data', data => message += data);
+
+    request.on('end', () => {
+      const additionalFlags = JSON.parse(message);
+      Object.assign(flags, additionalFlags);
+
+      lighthouse(database.url, flags, database.config).then(results => {
+        results.artifacts = undefined;
+        const id = database.saveData(flags, results);
+        response.writeHead(200);
+        response.end(`/?id=${id}`);
+      });
     });
-  });
+  } catch (err) {
+    if (err.code === 'MODULE_NOT_FOUND') {
+      response.writeHead(404);
+      response.end('404: Resource Not Found');
+    } else {
+      throw err;
+    }
+  }
+}
+
+/**
+ *  Store all the experiment data (flags and results) on hard disk.
+ *  Call this.clear() to delete all the temp files.
+ */
+class ExperimentDatabase {
+  constructor(url, config) {
+    this._url = url;
+    this._config = config;
+
+    this._root = fs.mkdtempSync(`${__dirname}/experiment-data`);
```

Comment:
It's not in CWD. It's under `lighthouse/lighthouse-cli/performance-experiment/`.
I tested it on YouTube.com, and the `results.json` is around 2.3 MB. It's pretty big already and could be bigger for some other websites. So I think we should store it in hard drive.
I addded a "-" to the end of the folder prefix. Thanks for the suggestion.